src/main/java/org/opensearch/ad/transport/BackPressureRouting.java

```diff
@@ -88,4 +84,20 @@ public boolean isMuted() {
     private void mute() {
         lastMuteTime = clock.millis();
     }
+
+    public int getMaxRetryForUnresponsiveNode() {
+        return maxRetryForUnresponsiveNode;
+    }
+
+    public void setMaxRetryForUnresponsiveNode(int maxRetryForUnresponsiveNode) {
+        this.maxRetryForUnresponsiveNode = maxRetryForUnresponsiveNode;
```

Comment:
Do the setters need to be thread safe?

Reply:
the method is called by addSettingsUpdateConsumer in NodeStateManager's constructor and addSettingsUpdateConsumer is a synchronized method.  So it is thread safe in our implementation.  I'll add a comment saying it is up to the client to make the method thread safe.